soh/soh/OTRGlobals.cpp

```diff
@@ -1509,3 +1522,90 @@ extern "C" s32 Randomizer_GetRandomizedItemId(GetItemID ogId, s16 actorId, s16 a
 extern "C" s32 Randomizer_GetItemIdFromKnownCheck(RandomizerCheck randomizerCheck, GetItemID ogId) {
     return OTRGlobals::Instance->gRandomizer->GetRandomizedItemIdFromKnownCheck(randomizerCheck, ogId);
 }
+
+extern "C" CustomMessageEntry Randomizer_GetCustomGetItemMessage(GetItemID giid, char* buffer, const int maxBufferSize) {
+    const CustomMessageEntry getItemText = CustomMessageManager::Instance->RetrieveMessage(Randomizer::getItemMessageTableID, giid);
+    return getItemText;
+}
+
+extern "C" int CustomMessage_RetrieveIfExists(GlobalContext* globalCtx) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    uint16_t textId = msgCtx->textId;
+    Font* font = &msgCtx->font;
+    char* buffer = font->msgBuf;
+    const int maxBufferSize = sizeof(font->msgBuf);
+    CustomMessageEntry messageEntry;
+    if (gSaveContext.n64ddFlag) {
+        if (textId == 0xF8) {
```

Comment:
`0xF8` feels like magic, maybe making a macro for it with a descriptive name would be helpful?

Reply:
Good idea, 0xF8 is an "unused" `textID` in vanilla (technically it's used but the box only contains the text of the address itself) that fits within the u8 type used for the `textID`s in `sGetItemEntries`.

Reply:
Added a file that contains an enum typedef for the raw textIDs that were present in the code.